packages/engine/src/ecs/functions/EntityFunctions.ts

```diff
@@ -15,7 +15,7 @@ export const removeEntity = (entity: Entity, immediately = false, world = Engine
     bitECS.removeEntity(world, entity)
   } else {
     removeAllComponents(entity, world)
-    addComponent(entity, EntityRemovedComponent, {})
+    addComponent(entity, EntityRemovedComponent, null)
```

Comment:
what's the reasoning for this? it will cause hasComponent checks to always return false

Reply:
it won't

Reply:
whats the reason for doing this instead of `true`

Reply:
true takes up space, and isn't necessary